### ISSUE TITLE
[UX] Form validation does not select the tab/fieldset that contains the required field

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#5504]
I came across this while working on #4302, and it is closely related to #5503.

When the form validation error is thrown, someone with no Backdrop/Drupal experience has to guess where the required field is located:

![image](https://user-images.githubusercontent.com/2423362/153695582-5eea2cd5-f611-467f-af06-1c02d8901663.png)

We should make it so that if the _first_ field mentioned in the form validation error is inside a vertical tab that is not the default/active one, or inside a fieldset that is collapsed by default, then we respectively select/focus the vertical tab, or expand the collapsed fieldset.

**Comment by klonos** [comment]
Done some quick research, and found some relevant posts/issues in d.org that may help:
- [Vertical tabs - show required fields indicator - highlight tabs when validation fails](https://www.drupal.org/forum/support/module-development-and-code-questions/2016-04-27/vertical-tabs-show-required-fields)
- [Correct vertical tab does not focus on form validation](https://www.drupal.org/project/drupal/issues/2911932) (Drupal core)
- [Add a user-defined "required" indicator to tabs containing required fields](https://www.drupal.org/project/vertical_tabs/issues/363279) (6.x vertical_tabs module)
- [Ensure visibility of invalid fields (JS)](https://www.drupal.org/project/field_group/issues/2787179) (8.x field_group module)
- [Vertical tabs required fields indicator does not work.](https://www.drupal.org/project/field_group/issues/2643558) (8.x field_group module)

**Comment by bugfolder** [comment]
Code looks good. Add "Fixes " to the first comment in the PR.

**Comment by bugfolder** [comment]
Here's a test where I would have expected it to work but it doesn't.

* Go to `node/add/page`.
* Enter a title and body.
* Click on "Authoring information" and enter "foobar" as the "authored by" name.
* Click "Save".

We do get the error "The username foobar does not exist" properly, but the selected vertical tab is "Publishing actions," not "Authoring information," as here:

![test](https://user-images.githubusercontent.com/29214084/154395386-18a426ec-1349-4157-acf1-f63bfdae39e5.jpg)

Maybe I'm expecting too much? It only selects for missing "required" fields, but not for general validation errors? (If so, should the latter be a follow-up issue?)

**Comment by bugfolder** [comment]
I do find that it works for the scenario described above: visit `admin/structure/types/manage/page` and leave the comments per page blank. 

If I leave out two required fields ("Title field label" in the first tab, "Comments per page" in the last tab), both missing fields are reported and the first tab with an error is selected (which seems appropriate).

**Comment by klonos** [comment]
> Maybe I'm expecting too much? It only selects for missing "required" fields...

That's right. And no, you are not expecting too much - any `:invalid` non-required field should also work. I'll adjust the PR and let you know when ready for another round of testing.

**Comment by klonos** [comment]
> If I leave out two required fields ("Title field label" in the first tab, "Comments per page" in the last tab), both missing fields are reported and the first tab with an error is selected (which seems appropriate).

During the Office Hours meeting today, I thought that perhaps we should somehow be indicating other "problematic" tabs too. I'll have a think about it and see if I can come up with something that works nicely.

**Comment by herbdool** [comment]
I closed this PR - it can be reopened once it has been rebased.
